src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -230,6 +230,29 @@ def _validate_non_args(self):
                         "yet has an ArgumentInterface interface."
                         "".format(str(symbol)))
 
+    def get_unlinked_datasymbols(self, ignore_precision=True):
+        '''
+        Create a list of the names of all of the Symbols in the table that
+        do not have a resolved interface. If ignore_precision is True then
+        those Symbols that are used to define the precision of other Symbols
+        are ignored. If no un-resolved Symbols are found then an empty list
+        is returned.
+
+        :param bool ignore_precision: whether or not to ignore Symbols that \
+                            are used to define the precision of other Symbols.
+
+        :returns: the names of those Symbols with deferred interfaces.
+        :rtype: list of str
+
+        '''
+        unlinked_symbols = set(self.unresolved_datasymbols)
```

Comment:
Don't we only need a set if ignore_precision is True? If so this should be moved inside the if clause and the else should just be self.unresolved_datasymbols.

Reply:
Oh yes. Tidied.